docs: fix selection interface in Code Block Docs (#766)

WaldemarLehner opened the pull request:
Updates Selection-Section of Docs to account for additional nesting.

Also added a tip-Block with an example on how to use multiple helpers at the same time.

WaldemarLehner commented:
Rendered Page: 
![image](https://github.com/motion-canvas/motion-canvas/assets/27686785/a4edc17c-eacb-4290-b348-6e926d0f52e0)

aarthificial commented on packages/docs/docs/components/code-block.mdx:
```diff
@@ -150,6 +177,19 @@ yield * codeRef().selection(lines(1), 1);
+
+You can also mix and match helpers by using the spread operator!
+
+You could select the 3rd line and a 10 character long word on the 5th line
```
```suggestion
You could select the 3rd line and a 10 characters long word on the 5th line
```

hhenrichsen replied:
This is grammatically correct as-is.

aarthificial replied:
I stand corrected

aarthificial reviewed (commented):
I think the original issue also mentions changing the order in the selection section so that we start by explaining the helpers and then move on to the raw arrays

hhenrichsen commented on packages/docs/docs/components/code-block.mdx:
```diff
@@ -109,20 +132,24 @@ yield view.add(
 It is often better, however, to use one of the three provided utility functions
-for selection. Though they all return a two by two array of numbers, they can be
-more descriptive than an array literal.
+for selection. Though they all return a array with one selection entry they can
```
nit: Though they all return **an** array with one selection entry

WaldemarLehner replied:
No longer relevant due to text changes

hhenrichsen commented on packages/docs/docs/components/code-block.mdx:
```diff
@@ -97,10 +96,34 @@ Code may be "selected" to call attention to it. The default behavior is to
-where `line` and `position` are zero indexed, and `position` is an insert
-position between characters. So `[[0, 5], [1, 2]]` will select the sixth
-character on the first line through to the second character on the second line.
+first created. Selections are specified using two `[line, position]` arrays
```
This section should come after we explain the helpers, since 90% of the time the helpers make more sense than the nested arrays.

WaldemarLehner commented:
Swapped order of Helpers and "raw" Arrays and also modified the text to have a logical flow around it.

I updated the rendered output in my first comment :)

WaldemarLehner commented:
@hhenrichsen i think we're waiting on your approval before merging :D

hhenrichsen commented:
LGTM!

WaldemarLehner commented:
@aarthificial this is good to merge :)
